### PR TITLE
Fix e2e code compatibility for aks-engine clusters

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
Fix e2e code compatibility for aks-engine clusters
* Remove unnecessary delete PIP function
* Use Linux selector for exec-agnhost-pod
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/service_annotations.go [hunk 1]
```diff
@@ -475,8 +475,6 @@ var _ = Describe("Service with annotation", Label(utils.TestSuiteLabelServiceAnn
 			By("Cleaning up service and public IP")
 			err := utils.DeleteService(cs, ns.Name, serviceName)
 			Expect(err).NotTo(HaveOccurred())
-			err = utils.DeletePIPWithRetry(tc, publicIP, tc.GetResourceGroup())
```
Comment (nilo19) [inline tests/e2e/network/service_annotations.go:478]: Why don't we need this?

Comment (lzhecheng) [reply tests/e2e/network/service_annotations.go:478]: I think we need this when explicityly called `WaitCreatePIP()`?

**Pull request merged**